packages/wdio-shared-store-service/src/service.ts

```diff
@@ -1,69 +1,89 @@
 import type { JsonCompatible, JsonPrimitive, Services, JsonArray, Capabilities } from '@wdio/types'
 
-import { getValue, setValue, setPort, setResourcePool, getValueFromPool, addValueToPool } from './client.js'
+import * as SharedStoreClient from './client.js'
 import { CUSTOM_CAP } from './constants.js'
 import type { SharedStoreServiceCapabilities } from './types.js'
-import type { GetValueOptions } from './types.js'
+import type { GetValueOptions, SharedStoreOptions } from './types.js'
+import { RequestError } from 'got'
+
+const { getValue, setValue, setPort, setResourcePool, getValueFromPool, addValueToPool, close } = SharedStoreClient
 
 export default class SharedStoreService implements Services.ServiceInstance {
     private _browser?: WebdriverIO.Browser | WebdriverIO.MultiRemoteBrowser
+    private _options?: SharedStoreOptions
+
+    constructor(options: SharedStoreOptions, caps: Capabilities.RemoteCapability) {
 
-    constructor(_: never, caps: Capabilities.RemoteCapability) {
-        const port = (
+        this._options = options ?? {}
+
+        this._options.port = this._options.port ?? (
             (caps as SharedStoreServiceCapabilities)[CUSTOM_CAP] ||
             ((caps as Capabilities.W3CCapabilities).alwaysMatch as SharedStoreServiceCapabilities)?.[CUSTOM_CAP] ||
             (Object.values(caps as Capabilities.MultiRemoteCapabilities)[0]?.capabilities as SharedStoreServiceCapabilities)[CUSTOM_CAP]
         )
 
-        if (!port) {
+        if (!this._options.port) {
             throw new Error('SharedStoreService: port not found in capabilities')
         }
 
-        setPort(port)
+        setPort(this._options.port)
+    }
+
+    async safeguard500(error: any) {
```

Comment:
Happy to and will do but can't really pinpoint exactly why it happens.

I've seen it happen often and it's crucial for my use not to break tests, thought that an option to enable might benefit not just me but other users. I suspect that it can be from high load on the machine running the server/tests.

Reply:
Maybe we should setup a retry mechanism for the client which uses `got`. Silently suppressing seems more like the last straw to do here.